Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerComponent.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI.h>
+
+namespace ROS2
+{
+    void ROS2SpawnerComponent::Activate()
+    {
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+
+        m_get_names_service = ros2Node->create_service<gazebo_msgs::srv::GetWorldProperties>(
+            "get_available_spawnable_names",
+            std::bind(&ROS2SpawnerComponent::GetAvailableSpawnableNames, this, AZStd::placeholders::_1, AZStd::placeholders::_2));
+
+        m_spawn_service = ros2Node->create_service<gazebo_msgs::srv::SpawnEntity>(
+            "spawn_entity", std::bind(&ROS2SpawnerComponent::SpawnRobot, this, AZStd::placeholders::_1, AZStd::placeholders::_2));
```

Comment:
It is safer to use a [lambda than std::bind](https://abseil.io/tips/108) and slightly more performant since it doesn't have to marshal the parameters to a tuple as part of binding them to unspecified std::bind result type.

```suggestion
            [this](const GetAvailableSpawnableNamesRequest request, GetAvailableSpawnableNamesResponse response) { GetAvailableSpawnableNames(request, response); });

        m_spawn_service = ros2Node->create_service<gazebo_msgs::srv::SpawnEntity>(
            "spawn_entity", [this](const SpawnEntityRequest request, SpawnEntityResponse response) { SpawnRobot(request, response); });
```